backend/src/plugins/authorization/authorization.plugin.ts

```diff
@@ -25,11 +27,16 @@ const auth: FastifyPluginAsync<Options> = async (fastify, opts) => {
 
   fastify.addHook(ControllerHook.ON_REQUEST, async (request, reply) => {
     try {
-      const isWhiteRoute = opts.routesWhiteList.some(
-        (route) => route === request.routerPath,
+      const whiteRoute = opts.routesWhiteList.find(
+        ({ route }) => route === request.routerPath,
       );
 
-      if (isWhiteRoute) {
+      const isWhiteRoute = Boolean(whiteRoute);
+      const isAllowedMethod = whiteRoute?.methods.includes(
```

Comment:
it can be a undefined now

do it as in the previous suggestion

Reply:
Returning is not an option, because it will break the code, so I had to have a second check inside `if (isWhiteRoute)`, to avoid `undefined`